pkg/controller/common/metrics.go

```diff
@@ -0,0 +1,64 @@
+package common
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	// DefaultBindAddress is the port for the metrics listener
+	DefaultBindAddress = ":8797"
+
+	// MachineConfigControllerPausedPoolKubeletCA logs when important config is being held up by pause
+	MachineConfigControllerPausedPoolKubeletCA = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "machine_config_controller_paused_pool_kubelet_ca",
+			Help: "important config is waiting in specified paused pool",
+		}, []string{"pool"})
+
+	metricsList = []prometheus.Collector{
+		MachineConfigControllerPausedPoolKubeletCA,
+	}
+)
+
+func RegisterMCCMetrics() error {
+	for _, metric := range metricsList {
+		err := prometheus.Register(metric)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// StartMetricsListener is metrics listener via http on localhost
+func StartMetricsListener(addr string, stopCh <-chan struct{}) {
```

Comment:
This looks very similar to [`daemon.StartMetricsListener`](https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/metrics.go#L100-L109). I wonder if there's an opportunity for consolidating the boilerplate of starting the listener, e.g.:

```golang
func StartMetricsListener(addr string, stopCh <-chan struct{}, register func() error) {
  // Boilerplate
  if err := register(); err != nil {
    // Handle error
  }
  // More boilerplate
}
```

Then the controller and daemon can call this and pass in their registration funcs.

Reply:
I've refactored StartMetricsListener to take a function argument as suggested, and adjusted daemon to call it from controller/common.